client/src/store/index.js

```diff
@@ -0,0 +1,105 @@
+import Vue from 'vue'
+import Vuex from 'vuex'
+import VuexPersistence from 'vuex-persist'
+import localforage from 'localforage'
+import Fuse from 'fuse.js'
+import { transformPublication } from './helpers'
+
+const vuexLocal = new VuexPersistence({
+  storage: localforage,
+})
+
+Vue.use(Vuex)
+
+const fuseOptions = {
+  // isCaseSensitive: false,
+  // includeScore: false,
+  // shouldSort: true,
+  // includeMatches: false,
+  // findAllMatches: false,
+  // minMatchCharLength: 1,
+  // location: 0,
+  // threshold: 0.6,
+  // distance: 100,
+  // useExtendedSearch: false,
+  keys: [
+    { name: 'abstract', weight: 0.5 },
+    { name: 'authors', weight: 1.2 },
+    { name: 'disciplines', weight: 1.8 },
+    { name: 'source', weight: 1.5 },
+    { name: 'title', weight: 2.0 },
+    { name: 'year', weight: 1.0 },
+  ],
+}
+
+let index = null
+
+export default new Vuex.Store({
+  state: {
+    loaded: false,
+    publications: [],
+    pageSize: 4,
+  },
+  mutations: {
+    HYDRATE_SINGLE_PUBLICATION: (state, publication) => {
+      const transformed = transformPublication(publication)
+      Vue.set(state, 'publications', [transformed])
+    },
+    HYDRATE_ALL_PUBLICATIONS: (state, publications) => {
+      publications = publications.map(transformPublication)
+      Vue.set(state, 'publications', publications)
+      Vue.set(state, 'loaded', true)
+
+      setTimeout(() => {
+        index = Fuse.createIndex(
+          fuseOptions.keys.map(({ name }) => name),
+          publications,
+        )
+      })
```

Comment:
Are you trying to do something in the lines of process.nextTick here? Then I'd probably go with requestAnimationFrame, setTimeout is somewhat undefined

Reply:
thats ok, the index is likely not needed until seconds after the timeout is scheduled, so there’s plenty of time and no need to worry about polyfills or render cycles. however, creating the index could be expensive and shouldn’t block initial time to first meaningful paint.